### ISSUE TITLE
[bug] HalfNormal distributions don't work with JitTrace_ELBO

### ISSUE THREAD

**Issue by reverendbedford** [pyro-ppl/pyro#2931]
### Issue Description
When I try to use `HalfNormal` distributions in my model I get the following error

```
  File "XX/.local/lib/python3.9/site-packages/torch/distributions/half_normal.py", line 55, in log_prob
    log_prob[value.expand(log_prob.shape) < 0] = -inf
RuntimeError: NYI: Named tensors are not supported with the tracer
```

Replacing the `HalfNormal` distributions with something else (say a `LogNormal` distribution) allows the model to run.

### Environment
 - Debian 
 - torch==1.9.0+cu111
 - pyro 1.7.0

### Code Snippet
Just replacing the `HalfNormal` is sufficient to avoid the error.

**Comment by fritzo** [comment]
@reverendbedford are you using any named tensors or packages that may be using named tensors? If not then I suspect this is an internal PyTorch bug and the error message is incorrect (since neither Pyro nor torch.distributions use named tensors).

**Comment by reverendbedford** [comment]
I am not using named tensors or any package using named tensors.  I  do suspect it's an internal error somewhere, as I can replace the `HalfNormal`s and everything works fine.  I can raise this over on PyTorch if you think the bug is there.

**Comment by fritzo** [comment]
Yes could you raise this issue on PyTorch, cc me and @neerajprad? We could add jit tests there to try to catch these earlier. It may be due to jitting expand or sth.) Thanks!
